### PR TITLE
Fix for ERSED being adjusted to fine ARD when fine is immediate release

### PR DESCRIPTION
**Pull request by srowlands-moj**
* Apply fix for ERSED being incorrectly adjusted to Fine ARD when fine is deemed immediate release.
* Enable SDSXX example 20 in calc engine scenarios
* Minor refactor to make BookingExtractionService method easier to follow and slightly more unit testable in the future

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt [hunk 1]
```diff
@@ -346,23 +378,30 @@ class BookingExtractionService(
...
-      extractionService.mostRecentOrNull(sentences.filterIsInstance<AFineSentence>(), SentenceCalculation::releaseDate)
-    val afineIsRelease = latestAFineRelease == latestReleaseDate
+      extractionService.mostRecentSentenceOrNull(sentences.filterIsInstance<AFineSentence>(), SentenceCalculation::releaseDate)
+    val afineIsRelease = latestAFineRelease?.sentenceCalculation?.releaseDate == latestReleaseDate
     if (latestEarlyReleaseSchemeEligibilitySentence != null && !afineIsRelease) {
       if (latestAFineRelease != null && latestEarlyReleaseSchemeEligibilitySentence.sentenceCalculation.earlyReleaseSchemeEligibilityDate!!.isBefore(
-          latestAFineRelease,
-        )
+          latestAFineRelease.sentenceCalculation.releaseDate,
+        ) && !latestAFineRelease.sentenceCalculation.isImmediateRelease()
```
Comment (james-baker-digital-justice) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:397]: just checking as I don't understand the logic. If there is another afine that is not an immediate release should we adjust to that?

Comment (srowlands-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:397]: Yes as there is time to be served on it still and it's a term of imprisonment 
It's a bit odd how it's treated differently in some scenarios but not in others

Comment (srowlands-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:397]: I plan on adding some tests fir the extracted methods today to make it clearer

Comment (srowlands-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:397]: Just thought I can get as in in now with existing coverage of checking the overall output

**Review by james-baker-digital-justice — DISMISSED** [review]
LGTM

**Review by james-baker-digital-justice — APPROVED** [review]
LGTM

**Pull request merged**